### PR TITLE
Adds Nullable validator

### PR DESCRIPTION
**Pull request by elijahbenizzy**
A few minor usabiility issues when I was messing with some Stitch Fix code.

## Changes
See commits.

## Testing

1.

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Python - local testing

- [ ] python 3.6
- [ ] python 3.7

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
This is very simple, happy to change names

File: hamilton/data_quality/default_validators.py [hunk 1]
```diff
@@ -374,6 +374,37 @@ def arg(cls) -> str:
         return 'mean_in_range'
 
 
+class NoNonesAllowedValidatorAllTypes(BaseDefaultValidator):
+
+    def __init__(self, allow_none: bool, importance: str):
+        super(NoNonesAllowedValidatorAllTypes, self).__init__(importance)
+        self.allow_none = allow_none
```
Comment (skrawcz) [inline hamilton/data_quality/default_validators.py:381]: The parameter name here is inconsistent with the class name; we want it to be obvious which validator maps to the kwarg. I.e. so either `no_nones_allowed` is the kwarg, or the class name is `AllowNoneValidatorAllTypes`.

File: hamilton/graph.py [hunk 2]
```diff
@@ -17,56 +17,16 @@
 from hamilton import node
 from hamilton.node import NodeSource, DependencyType
 from hamilton import base
+from hamilton.type_utils import custom_subclass_check
```
Comment (skrawcz) [inline hamilton/graph.py:20]: inconsistent with our style.

This should be `from hamilton import type_utils` -- and then where the function is used, prefixed with the module.

**Review by skrawcz — CHANGES_REQUESTED** [review]

File: hamilton/graph.py (hunk 2 above)
Comment (elijahbenizzy) [reply hamilton/graph.py:20]: We need an automated checker...

File: hamilton/data_quality/default_validators.py (hunk 1 above)
Comment (elijahbenizzy) [reply hamilton/data_quality/default_validators.py]: Sure, although I'm not sure there's value in consistency over self-description here.

Comment (elijahbenizzy) [reply hamilton/data_quality/default_validators.py]: Sure, although I'm not sure there's value in consistency over self-description here.

**Review by skrawcz — APPROVED** [review]
you going to squash merge with a nice commit?

**Comment by elijahbenizzy** [comment]
> you going to squash merge with a nice commit?

Going to quickly consolidate into two commits

> you going to squash merge with a nice commit?

Nope I'll turn this into two commits then commit them both -- should have been two PRs

**Pull request merged**